### ISSUE TITLE
Edits to the File menu after Version 0.1.0 including sjmisc for climatic

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2244]
First this is a big congratulations to all of us.  I saw the excitement when we "released" the very first version of R-Instat in August in Bagamoyo.  It then became clear we are really potentially doing something pretty outstanding.  For me - since then - I was excited by the reactions from our Lesotho visitors (Met Service) where I can see it can clearly change their working practices.  And I know some of you, in Maseno, were happy when you gave a demonstration recently.  Well they ain't seen nuthin yet!  Just think what it is now, and where it can be - and pretty soon.

But this is a message about issues.  Here I give you my personal list of issues that I would like to resolve in the near future.  Most will not be the only, or even the most major tasks for now - we need to multi-task, but I hope we can resolve some of these items, mostly sort of "on the side"  
I know there will mainly soon be a major overhaul, with sub-dialogues being handled differently, and the Instat object more clearly working.  And we DO need to work on the climatic menu - and in November.  But, in parallel, we have a range of issues - some pretty simple - that I hope we can start to address.  In this list, I am going menu-by-menu.  So first I am on the File menu.

1) For importing it was really excellent moving to our use of the rio package.  And Danny seems to have checked that we can now read ods files from Open Office.  We can read also from SPSS and Stata, but how do we also read in the value labels?  This would make us very popular.  There is (of course) discussion on the web, for example [http://stackoverflow.com/questions/19914962/importing-values-and-labels-from-spss-with-memisc](url)  (See also below - it may be that haven or sjmisc are more appropriate.) 

We are also still looking to resolve some problems importing "awkward" sets of data - we had one in Bagomoyo?  And we are not quite clear what "awkward" means, except that R-Instat doesn't like them!

2) Will we do "Convert"?  This would be low on my priority list, but it may be easy.

3) Now the big one, which is more general than the file menu.  We seem to be able to handle "long" - or at least longish sets of data.  **We need to do better with data sets that are wide.**  We can read data sets in easily from the library.  I now have input the following sets into my version 0.1.0 at the same time:
a) The little TestProb file - with just 10 rows.
b| The diamonds dataset from ggplot2, with 50,000 rows and 10 columns
c) Birthdays from Mosaicdata with 372,864 rows and 7 columns.  This is interesting, because it has dates (like our climatic data) and in 2 ways, i.e. a date column and also the year, month, day.  I also see we recognise the date type in our column attributes, but express it in a complicated way.  We don't have "(d)" in the grid view.
d) HELPfull data also from MosaicData which has 1472 rows and 788 columns - so one of the type we need to be able to handle well.
e) rockArt from DAAG with 103 rows and 641 columns.
I hope we can improve our handling of wide data sets for the next release.  I think one key issue is the way that dialogues are populated?

There is an important set of issues (in addition to being able to load dialogues efficiently) that are related to our need to handle "wide" sets efficiently.  We already have the makings of a great filtering system on rows, and we will need it also on columns - so we don't have to work with them all - either in the grid or generally.  We also need to be able to review and edit column names efficiently. I return to this point below.

4) For reference here is to add further types of data into our "library".  I return to this in a separate issue on the Help system.  In our reading of data from the library I really like the second help button!  By the way, a small issue there.  Once we use that help, would it be easy to return the focus to the same dataset.  (Currently it becomes unselected.)  Then we can just press OK to use it. 

Danny has requested a different "scheme" where made by individual dialogues.  So that's what I'll do now.

**Comment by rdstern** [comment]
From the list above it would be good to make progress on the importing (and exporting) to the popular statistics packages - particularly SPSS and Stata.  In particular we do need to be able to import from SPSS (and hopefully Stata) with value labels.  This seems to be recent work - as usual by Hadley Wickham!  It uses haven, which (I think) is also used by us through rio.  What this says is as follows:

(From SPSS) "You always get a data frame, date times are converted to corresponding R classes and labelled vectors are returned **as new labelled class**. You can easily coerce to factors or replace labelled values with missings as appropriate. If you also use dplyr, you'll notice that large data frames are printed in a convenient way."

If this isn't the route - and anyway it is interesting - the other package that seems relevant is called **sjmisc**.

It would be great if someone could take up this challenge.

Roger

**Comment by rdstern** [comment]
Reading further I think we should examine the **sjmisc** package and now is a good time!  Here is a document on this package.  It discusses the haven way of importing value labels and considers that it does better - through using attributes!  It also discusses the issue of multiple missing value codes.  It all looks interesting.  

And it has a partner package called **sjplot** that I think we may be using soon.  I have a feeling that with these 2 packages we might have found another real "goodie".

**Comment by rdstern** [comment]
I refer back to the sjmisc package for a possibly curious reason.  In the climatic data from CLIMSOFT the station records have the starting date and the end date.  If the station is still open the end date is coded as 31/12/9999.  When you make this into a date it comes as a missing value.  It is a sort of missing value, but different from not knowing the end date.  This is a climatic example where there is more than one type of missing or special codes.  Another example could be the start of the rains where there is no start in some years.  In the old Instat I have coded this as zero, but perhaps we could have a special code for this.

So, this is a bit like the special missing value codes you get in surveys, where you could have missing, refused, not at home, etc.  This is what sjmisc claims to deal with.  It would be good to investigate soon.

**Comment by dannyparsons** [comment]
Good to check on multiple missing dates in sjmisc - but I don't see why 31/12/9999 comes out as missing. Only if these values are in a different to the rest of the column perhaps? But doing `as.Date("31/12/9999", format = "%d/%m/%Y")` works perfectly fine.
